ui/src/app/machines/components/ActionFormWrapper/ActionFormWrapper.tsx

```diff
@@ -107,7 +107,12 @@ export const ActionFormWrapper = ({
         case "tag":
           return <TagForm setSelectedAction={setSelectedAction} />;
         case "test":
-          return <TestForm setSelectedAction={setSelectedAction} />;
+          return (
+            <TestForm
+              hardwareType={selectedAction.hardwareType}
```

Comment:
The `hardwareType` prop here feels right to me, as the `TestForm` can have different groups of tests selected depending on hardware, *however* I'm not entirely sure about setting `hardwareType` using `setSelectedAction`. Passing `setSelectedAction` down so many components I also don't love, but perhaps this just needs to be reworked later in redux. Any thoughts?